git/options.go

```diff
@@ -39,22 +67,39 @@ type CheckoutOptions struct {
 	// SemVer tag expression to checkout, takes precedence over Tag.
 	SemVer string `json:"semver,omitempty"`
 
-	// Commit SHA1 to checkout, takes precedence over Tag and SemVer,
-	// can be combined with Branch with some Implementations.
+	// Commit SHA1 to checkout, takes precedence over Tag and SemVer.
+	// If supported by the client, it can be combined with Branch.
 	Commit string
+}
 
-	// RecurseSubmodules defines if submodules should be checked out,
-	// not supported by all Implementations.
-	RecurseSubmodules bool
+// CommitOptions provides options to configure a Git commit operation.
+type CommitOptions struct {
+	// Signer can be used to sign a commit using OpenPGP.
+	Signer *openpgp.Entity
+	// Files contains file names mapped to the file's content.
+	// Its used to write files which are then included in the commit.
+	Files map[string]io.Reader
+}
```

Comment:
With the function options, this struct can now be private. Options can be set via their function options.

Reply:
But how would external packages that implement `Commit` actually access the struct?

Reply:
The public API to configure `CommitOption` is `WithSigner(value)` and `WithFiles(value)`. They don't need to directly use the `CommitOptions`. We have this pattern everywhere.

```go
r.Commit(c, WithSigner(s), WithFiles(f))
```

Reply:
Oh, this is in the git package and you need it in the other packages.

Reply:
How would they get access to the underlying signer or files to be used in the actual implementation of `Commit()`? `pkg/git/gogit` can't access `pkg/git.commitOptions`

Reply:
Looks like an exception to the pattern because of the dependency of others on it. Don't think we can do anything about it. Let it be.